database/engine/rrdengine.c

```diff
@@ -99,6 +273,34 @@ void read_extent_cb(uv_fs_t* req)
         debug(D_RRDENGINE, "LZ4 decompressed %u bytes to %d bytes.", payload_length, ret);
         /* care, we don't hold the descriptor mutex */
     }
+    {
+        uint8_t xt_is_cached = 0, xt_is_inflight = 0;
+        unsigned xt_idx;
+        struct extent_info *extent = xt_io_descr->descr_array[0]->extent;
+
+        xt_is_cached = !lookup_in_xt_cache(wc, extent, &xt_idx);
+        xt_is_inflight = check_bit(wc->xt_cache.inflight_bitmap, xt_idx);
+        if (xt_is_cached && xt_is_inflight) {
```

Comment:
xt_idx is undefined if xt_is_cache is false

Reply:
It should be safe in practice but it's quite ugly so I'm fixing it.